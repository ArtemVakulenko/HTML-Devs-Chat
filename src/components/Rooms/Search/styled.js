import styled from 'styled-components';

export const StSearchContainer = styled.div`
  padding: 18px 25px;
  display: flex;
  justify-content: center;
  align-items: center;
`;
