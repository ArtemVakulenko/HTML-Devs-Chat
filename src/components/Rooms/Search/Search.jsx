import React from 'react';
import PropTypes from 'prop-types';
import Input from 'src/components/_common/Input';
import { StSearchContainer } from './styled';

const Search = ({ value, onChange }) => (
  <StSearchContainer>
    <Input
      value={value}
      onChange={onChange}
      placeholder='search'
    />
  </StSearchContainer>
);

export default Search;

Search.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};
