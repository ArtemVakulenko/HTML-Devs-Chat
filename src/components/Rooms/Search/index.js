import { connect } from 'react-redux';
import { searchValue } from 'store/mainPage/selectors';
import { setSearchValue } from 'store/mainPage/actions';
import Search from './Search';

const mapStateToProps = (state) => ({
  value: searchValue(state),
});
const mapDispatchToProps = {
  onChange: setSearchValue,
};
export default connect(mapStateToProps, mapDispatchToProps)(Search);
