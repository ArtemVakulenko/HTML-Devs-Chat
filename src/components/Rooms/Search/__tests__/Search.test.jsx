import React from 'react';
import Search from '../Search';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
  })),
}));

jest.mock('../../../hooks', () => ({
  useTheme: jest.fn().mockImplementation(() => ({
    colors: {
      dark: {
        color: '',
        input: '',
        placeholder: '',
      },
    },
    theme: 'dark',
  })),
}));
describe('Search', () => {
  const props = {
    value: 'MockValue',
    onChange: jest.fn(),
  };
  it('Should match snapshot', () => {
    const component = global.shallow(<Search {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have input', () => {
    const component = global.mount(<Search {...props} />);
    expect(component.find('input')).toHaveLength(1);
  });
  it('Should have called onchange', () => {
    const newValue = 'MockValueNew';
    const component = global.mount(<Search {...props} />);
    component.find('input').simulate('change', { target: { value: newValue } });
    expect(component.props().onChange).toHaveBeenCalledWith(newValue);
  });
});
