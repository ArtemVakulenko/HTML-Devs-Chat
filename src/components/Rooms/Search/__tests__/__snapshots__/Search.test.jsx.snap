// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Should match snapshot 1`] = `"<div class=\\"styled__StSearchContainer-bRxinB cbrvhe\\"><input type=\\"text\\" value=\\"MockValue\\" class=\\"styled__StInput-ifVcmW bBpwQJ\\"/></div>"`;
